Microsoft.Azure.Cosmos/src/CosmosClient.cs

```diff
@@ -1042,6 +1088,36 @@ internal DatabaseProperties PrepareDatabaseProperties(string id)
                options: requestOptions);
         }
 
+        private async Task InitializeContainerAsync(string databaseId, string containerId, CancellationToken cancellationToken = default)
+        {
+            ContainerInternal container = (ContainerInternal)this.GetContainer(databaseId, containerId);
+            IReadOnlyList<FeedRange> feedRanges = await container.GetFeedRangesAsync(cancellationToken);
+            List<Task> tasks = new List<Task>();
+            foreach (FeedRange feedRange in feedRanges)
+            {
+                tasks.Add(CosmosClient.InitializeFeedRangeAsync(container, feedRange, cancellationToken));
+            }
+
+            await Task.WhenAll(tasks);
+        }
+
+        private static async Task InitializeFeedRangeAsync(ContainerInternal container, FeedRange feedRange, CancellationToken cancellationToken = default)
+        {
+            // Do a dummy querry for each Partition Key Range to warm up the caches and connections
+            string guidToCheck = Guid.NewGuid().ToString();
+            QueryDefinition queryDefinition = new QueryDefinition($"select * from c where c.id = '{guidToCheck}'");
+            using (FeedIterator feedIterator = container.GetItemQueryStreamIterator(feedRange,
+                                                                                    queryDefinition,
+                                                                                    continuationToken: null,
+                                                                                    requestOptions: new QueryRequestOptions() { }))
+            {
+                while (feedIterator.HasMoreResults)
+                {
+                    await feedIterator.ReadNextAsync(cancellationToken);
```

Comment:
This is the stream iterator. You need to check the status code. It should not throw for 404 or other scenarios. 
u
```suggestion
                    using ResponseMessage response = await feedIterator.ReadNextAsync(cancellationToken);
                    response.EnsureSuccessStatusCode();
```